editor/src/components/canvas/remix/utopia-remix-root-component.tsx

```diff
@@ -75,7 +93,86 @@ function useGetRouteModules(basePath: ElementPath) {
     }
 
     return routeModulesResult
-  }, [defaultExports, remixDerivedDataRef, projectContentsRef])
+  }, [
+    defaultExports,
+    metadataContext,
+    remixDerivedDataRef,
+    projectContentsRef,
+    fileBlobsRef,
+    hiddenInstancesRef,
+    displayNoneInstancesRef,
+  ])
+}
+
+function useGetRoutes() {
+  const routes = useEditorState(
+    Substores.derived,
+    (store) => store.derived.remixData?.routes ?? [],
```

Comment:
`MemoryRouter` has an invariant that it can only be passed a non-empty array of routes, so we should check the length of `routes` before the router is created later on in this file